### PR TITLE
Fix extraction of SNV from VCF when adjacent to synonym for SNV

### PR DESCRIPTION
**Pull request by helensch**

### CODE REVIEW

**Comment by at7** [comment]
Reducing the query region to 1 for SNPs definitely works. However, I don't know why we consider synonyms at all (https://github.com/Ensembl/ensembl-variation/blob/master/modules/Bio/EnsEMBL/Variation/VCFCollection.pm#L1333). We should only match by allele and location. But this is only a comment for future improvements.

File: modules/t/VCFCollection_synonym.t [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+my $v  = $va->fetch_by_name('rs7569578');
+$multi->hide('variation', 'variation_synonym');
+
+# Temporarily add a synonym that is located in the VCF 1 base before
+# the variant of interest rs7569578
+$v->add_synonym('Archive dbSNP', 'synonym_1');
+$va->store_synonyms($v);
+my ($vf) = @{$v->get_all_VariationFeatures};
+my $gts = $coll->get_all_SampleGenotypeFeatures_by_VariationFeature($vf);
+ok($gts && scalar @$gts == 3, "get_all_SampleGenotypeFeatures_by_VariationFeature with nearby synonym - count 3");
```
[the remark this answers is not in the document]
Comment (helensch) [reply modules/t/VCFCollection_synonym.t]: Thanks for the suggestion.  This  was based on test in VCFCollection.t. I have created a JIRA ticket to improve the test for both ENSVAR-2733

**Comment by at7** [comment]
Thank you. I can merge the PR. Can you please create the PR also against postreleasefix/99?

**Comment by helensch** [comment]
I have created a ticket ENSVAR-2734 to review synonym use in VCFCollection.pm and improved matching

**Comment by helensch** [comment]
Yes, I will create a PR against postreleasefix/99

**Pull request merged**